server/consumer.go

```diff
@@ -265,12 +265,21 @@ func (mset *Stream) AddConsumer(config *ConsumerConfig) (*Consumer, error) {
 
 	// Make sure any partition subject is also a literal.
 	if config.FilterSubject != "" {
-		// Make sure this is a valid partition of the interest subjects.
-		if !mset.validSubject(config.FilterSubject) {
-			return nil, fmt.Errorf("consumer filter subject is not a valid subset of the interest subjects")
+		// If this is a direct match for the streams only subject clear the filter.
+		mset.mu.RLock()
+		if len(mset.config.Subjects) == 1 && mset.config.Subjects[0] == config.FilterSubject {
+			config.FilterSubject = _EMPTY_
```

Comment:
nice, discovered this annoyance too when I did a subject based client POC

Reply:
But do we plan maybe in the future to allow the modification of a stream as to be able to add a non overlapping subject to it? Say stream "STREAM" captures "foo" at the moment, then later the user decides to add "bar" to this stream. If we were to want to allow that in the future, would this change here not prevent that feature from ever existing?

Reply:
If decide to allow that we can update to correctly work but I do not see us doing that anytime soon.